### PR TITLE
[editor] HuggingFaceText2SpeechTransformerPromptSchema

### PR DESCRIPTION
**Pull request by rholinshead**
# [editor] HuggingFaceText2SpeechTransformerPromptSchema

Screenshot 2024-01-09 at 11 54 12 PM

### CODE REVIEW

File: python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceText2SpeechTransformerPromptSchema.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  // See https://huggingface.co/transformers/v4.11.3/main_classes/model.html and
+  // https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py
+
+  input: {
+    type: "string",
+  },
+  model_settings: {
+    type: "object",
+    properties: {
+      torch_dtype: {
```
Comment (saqadri) [inline python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceText2SpeechTransformerPromptSchema.ts:13]: Do we need a "model" property as well to be able to specify the model ID?

**Review by saqadri — APPROVED** [review]

File: python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceText2SpeechTransformerPromptSchema.ts (hunk 1 above)
Comment (rholinshead) [reply python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceText2SpeechTransformerPromptSchema.ts:13]: It's not currently supported but if we'll be adding it then I can add it here to require one less change later?

Comment (saqadri) [reply python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceText2SpeechTransformerPromptSchema.ts:13]: I can add it too. I was thinking all HF model setting schemas accept a "model". Then the model parsers use it as appropriate.

Comment (rholinshead) [reply python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceText2SpeechTransformerPromptSchema.ts:13]: I can add to each in next PR

Comment (rossdanlm) [reply python/src/aiconfig/editor/client/src/shared/prompt_schemas/HuggingFaceText2SpeechTransformerPromptSchema.ts:13]: > I was thinking all HF model setting schemas accept a "model". Then the model parsers use it as appropriate.

This also fits in with this general task of separating out the pipeline generation params from the actual completion params: https://github.com/lastmile-ai/aiconfig/issues/824

cc @Ankush-lastmile

**Pull request merged**